test/Infrastructure/LeanCode.AuditLogs.Tests/AuditLogsFilterTests.cs

```diff
@@ -0,0 +1,175 @@
+
+    private readonly IServiceProvider serviceProvider;
+    private readonly ITestHarness harness;
+    private static readonly TestEntity TestEntity = new() { Id = SomeId };
+
+    public AuditLogsFilterTests()
+    {
+        TestTimeProvider.ActivateFake(new DateTimeOffset(2023, 10, 6, 11, 0, 3, 0, TimeSpan.Zero));
+        var collection = new ServiceCollection();
+        collection
+            .AddOpenTelemetry()
+            .ConfigureResource(r => r.AddService("AuditLogsFilterTests", serviceInstanceId: Environment.MachineName))
+            .WithTracing(builder =>
+            {
+                builder
+                    .AddProcessor<IdentityTraceAttributesFromBaggageProcessor>()
+                    .AddSource("MassTransit")
+                    .AddLeanCodeTelemetry();
+            });
+        collection.AddDbContext<TestDbContext>();
+        collection.AddTransient<IAuditLogStorage, StubAuditLogStorage>();
+        collection.AddMassTransitTestHarness(ConfigureMassTransit);
+
+        serviceProvider = collection.BuildServiceProvider();
+        harness = serviceProvider.GetRequiredService<ITestHarness>();
+    }
+
+    private static void ConfigureMassTransit(IBusRegistrationConfigurator cfg)
+    {
+        cfg.AddConsumersWithDefaultConfiguration(
+            new[] { typeof(Consumer).Assembly, typeof(AuditLogsConsumer).Assembly },
+            typeof(DefaultConsumerDefinition<>)
+        );
+
+        cfg.UsingInMemory(
+            (ctx, busCfg) =>
+            {
+                busCfg.ConfigureEndpoints(ctx, new DefaultEndpointNameFormatter("InMemory"));
+                busCfg.ConnectBusObservers(ctx);
+            }
+        );
+    }
+
+    [Fact]
+    public async Task Publishes_entity_changes_to_bus()
+    {
+        await harness.Start();
+        var consumerHarness = harness.GetConsumerHarness<Consumer>();
+
+        await harness.Bus.Publish(new TestMsg());
+        (await consumerHarness.Consumed.Any<TestMsg>()).Should().BeTrue();
+
+        harness.Published
+            .Select<AuditLogMessage>()
+            .Should()
+            .ContainSingle()
+            .Which.Context.Message.Should()
+            .BeEquivalentTo(
+                new
+                {
+                    EntityChanged = new
+                    {
+                        Ids = new string[] { SomeId },
+                        Type = typeof(TestEntity).FullName,
+                        EntityState = "Added",
+                        Changes = JsonSerializer.SerializeToDocument(TestEntity, Options),
+                    },
+                    ActionName = typeof(Consumer).FullName,
+                    ActorId = null as string,
+                    DateOccurred = Time.NowWithOffset,
+                },
+                opt => opt.ComparingByMembers<JsonElement>()
+            )
+            .And.Subject.Should()
+            .Match((s) => s.As<AuditLogMessage>().SpanId != null && s.As<AuditLogMessage>().TraceId != null);
```

Comment:
```suggestion
            .Match(s => s.As<AuditLogMessage>().SpanId is not null && s.As<AuditLogMessage>().TraceId is not null);
```

Although i'm not sure about the `is not` change (might be related to this being an `Expression`, which I don't know if it is).

Reply:
It is. I tried to use `is not null` here but unfortunately it is not supported :disappointed: